sympy/solvers/tests/test_inequalities.py

```diff
@@ -314,5 +314,10 @@ def test_issue_8974():
     assert isolve(oo > x, x) == And(-oo < x, x < oo)
 
 
+def test_issue_10198():
+    assert reduce_inequalities(-1+1/abs(1/x-1)<0)== And(-oo < x, x < 0) or And(0 < x, x < 1/2)
```

Comment:
nits: please write as `reduce_inequalities(-1 + 1/abs(1/x - 1) < 0) == ...` (notice added whitespace). Also, there must be something wrong because the right side should be `Or(And(-oo < x, x < 0), And(0 < x, x < S(1)/2))`. Notice the difference below:

```
>>> Or(And(-oo < x, x < 0), And(0 < x, x < S(1)/2))
Or(And(-oo < x, x < 0), And(0 < x, x < 1/2))

>>> And(-oo < x, x < 0) or And(0 < x, x < S(1)/2)
And(-oo < x, x < 0)
```

Why did the tests pass with the expression you have? Not sure.